### DISCUSSION TITLE
How to manipulate volume and bucket metadata ?

### DISCUSSION THREAD

**Discussion by azro352** [apache/ozone#6684] [FAQ] [answered]
Hi, 

The [section](https://ozone.apache.org/docs/edge/security/securityacls.html) about the different rights mentions `metadata` for volumes and buckets

> Read – Allows the user to read the metadata of a Volume and Bucket and ...
> Write - Allows the user to write the metadata of a Volume and Bucket and ...

But I can't find any other mention of this in the documentation or in the `ozone sh bucket --help` (looked there because we can see about them in the comand output)

[ozone cli](https://ozone.apache.org/docs/edge/interface/cli.html)

```bash
$ ozone sh volume info /vol1
{
  "metadata" : { },
  "name" : "vol1",
  "admin" : "hadoop",
  "owner" : "hadoop",
  "creationTime" : "2020-07-28T12:31:50.112Z",
  "modificationTime" : "2020-07-28T12:31:50.112Z",
  "acls" : [ {
    "type" : "USER",
    "name" : "hadoop",
    "aclScope" : "ACCESS",
    "aclList" : [ "ALL" ]
  }, {
    "type" : "GROUP",
    "name" : "users",
    "aclScope" : "ACCESS",
    "aclList" : [ "ALL" ]
  } ],
  "quota" : 1152921504606846976
}

$ ozone sh bucket info /vol1/bucket1
{
  "metadata" : { },
  "volumeName" : "vol1",
  "name" : "bucket1",
  "storageType" : "DISK",
  "versioning" : false,
  "creationTime" : "2020-07-28T13:14:45.091Z",
  "modificationTime" : "2020-07-28T13:14:45.091Z",
  "encryptionKeyName" : null,
  "sourceVolume" : null,
  "sourceBucket" : null
}
```

Is there anywhere that explains how to read/write these metadata ?

**Comment by ChenSammi** [comment, marked as answer]
Currently the metadata of volume/bucket/key only supports update by OzoneManager.  There is no CLI exposed to manipulate it.

**Reply by azro352** [reply]
Can you share some examples of these metadata ? Or maybe somewhere in the code I can see about them ?

**Reply by ChenSammi** [reply]
One metadata example is OzoneConsts.GDPR_FLAG.
